### PR TITLE
SPRacingH7 install instructions

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 1]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+#. Flash the binary using dfu-util -D AP_2MB.bin -s 0x90100000:0x200000
+#. Verify the flash. The dfu-util command below copies the contents of the flash back to the computer, the diff command will tell you if the contents are identical or different. Do not attempt to fly if diff doesn't say the files are identical - retry.
+
+dfu-util -U AP_2MB-VERIFY.bin -s 0x90100000:0x200000
+diff -sb AP_2MB.bin AP_2MB-VERIFY.bin
+
+#. Power off, install an SD card (.note: you *must* install an SD card, the firmware will not boot without it), power on
+#. Configure the board as normal using Mission Planner
+
+At this point you should have working firmware on the board. If you want to load new firmware you will need to follow steps 2-7 again (you cannot use Mission Planner to load firmware). If you are certain that you will never want to load betaflight on the board then you can install the ardupilot bootloader.
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:167]:
```suggestion
At this point you should have working firmware on the board. If you want to load new firmware you will need to follow steps 2-7 again (you cannot use Mission Planner to load firmware). If you are certain that you will never want to load betaflight on the board then you can install the ArduPilot bootloader.
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 2]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+
+Installing the ardupilot bootloader
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:169]:
```suggestion
Installing the ArduPilot bootloader
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 3]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+-----------------------------------
+
+.. note::
+  Caution - installing the ardupilot bootloader is a one-way operation. You cannot restore the board to factory configuration or load betaflight after this step - you would have to return the board to Seriously Pro to be re-flashed with factory firmware
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:173]:
```suggestion
  Caution - installing the ArduPilot bootloader is a one-way operation. You cannot restore the board to factory configuration or load betaflight after this step - you would have to return the board to Seriously Pro to be re-flashed with factory firmware
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 4]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+
+If you are certain that you only want to use ardupilot on the board then flashing the ardupilot bootloader enables much simpler subsequent upgrades.
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:175]:
```suggestion
If you are certain that you only want to use ArduPilot on the board then flashing the ardupilot bootloader enables much simpler subsequent upgrades.
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 5]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+
+#. You must initially have a working version of ardupilot installed on the board - follow the steps above.
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:177]:
```suggestion
#. You must initially have a working version of ArduPilot installed on the board - follow the steps above.
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 6]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+#. You now must remove the copy protection on the internal flash. This is a destructive operation requiring complete erasure of the flash. Ardupilot provides support to make this easy. Set:
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:178]:
```suggestion
#. You now must remove the copy protection on the internal flash. This is a destructive operation requiring complete erasure of the flash. ArduPilot provides support to make this easy. Set:
```

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 7]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+- :ref:`BRD_OPTIONS` = 16
+#. Power off and power on the board. The board will not appear to boot but the flash sector is being erased. Wait a few seconds and then power off the board.
+#. Hold down the ``boot`` button (boot *not* bind this time) and power on the flight controller. This will put the board in dfu mode.
+#. Download the ardupilot bootloader, e.g. https://github.com/ArduPilot/ardupilot/blob/master/Tools/bootloaders/SPRacingH7_bl.bin
```
Comment (nkruzan) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:182]:
```suggestion
#. Download the ArduPilot bootloader, e.g. https://github.com/ArduPilot/ardupilot/blob/master/Tools/bootloaders/SPRacingH7_bl.bin
```

**Review by nkruzan — COMMENTED** [review]
Just some nitpicks. Thanks!

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 8]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+
+At this point you should have working firmware on the board. If you want to load new firmware you will need to follow steps 2-7 again (you cannot use Mission Planner to load firmware). If you are certain that you will never want to load betaflight on the board then you can install the ArduPilot bootloader.
+
+Installing the ArduPilot bootloader
+-----------------------------------
+
+.. note::
+  Caution - installing the ArduPilot bootloader is a one-way operation. You cannot restore the board to factory configuration or load betaflight after this step - you would have to return the board to Seriously Pro to be re-flashed with factory firmware
+
+If you are certain that you only want to use ArduPilot on the board then flashing the ardupilot bootloader enables much simpler subsequent upgrades.
```
Comment (Hwurzburg) [inline common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:175]:
```suggestion
**If you are certain that you only want to use ArduPilot on the board**, then flashing the ardupilot bootloader enables much simpler subsequent upgrades.
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
biggest question is "what does a Windows user do??"

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 9]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+#. Use dd to pad the binary to 2MB:
+
+dd if=/dev/zero ibs=1k count=2048 of=AP_2MB.bin
+dd conv=notrunc if=arducopter_extf.bin of=AP_2MB.bin
+
+#. Put the board into SSBL dfu mode - power off, hold BIND (not BOOT), power on - LED flashes fast, release BIND, LED flashed slow - DFU mode enabled
+#. Flash the binary using dfu-util -D AP_2MB.bin -s 0x90100000:0x200000
+#. Verify the flash. The dfu-util command below copies the contents of the flash back to the computer, the diff command will tell you if the contents are identical or different. Do not attempt to fly if diff doesn't say the files are identical - retry.
+
+dfu-util -U AP_2MB-VERIFY.bin -s 0x90100000:0x200000
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:161]: TBF these are existing issues for SPRacing users, not sure we can paper over the cracks

File: common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst [hunk 10]
```diff
@@ -136,3 +136,53 @@ Upload ArduPilot to the board
...
+Loading firmware using SSBL
+---------------------------
+
+The SPRacing series of boards come pre-installed with a proprietary bootloader on the internal flash and require the use of a second stage bootloader to load further firmware. There are a couple of options to load firmware with these boards, but whichever option you choose you will need to initially load ArduPilot using SSBL. Please follow the instructions at https://github.com/spracing/ssbl in order to load SSBL onto your board. Once SSBL is loaded please follow the PX4 instructions to load ArduPilot onto the board https://github.com/spracing/ssbl#px4-installation-to-external-flash but instead of using PX4 firmware please use the arducopter_extf.bin firmware image. A summary of the steps follows:
+
+#. Download https://github.com/spracing/ssbl/releases and install SSBL to external flash following https://github.com/spracing/ssbl#installation-to-external-flash
+#. Download the latest ArduPilot external flash binary, for instance https://firmware.ardupilot.org/Copter/latest/SPRacingH7/arducopter_extf.bin
+#. Use dd to pad the binary to 2MB:
+
+dd if=/dev/zero ibs=1k count=2048 of=AP_2MB.bin
```
[the remark this answers is not in the document]
Comment (tridge) [reply common/source/docs/common-loading-firmware-onto-chibios-only-boards.rst:154]: yes, if it needs padding we should pad when we build, not require user to pad

**Pull request merged**